transcom/mymove
# INT B-20011 B-20013 B-20014 HQ Role Queues, Searching, and Permissions

## [B-20011](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20011) & [B-20013](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20013) & [B-20014](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20014)

## Summary

Adds the following changes for the HQ role:
- Ability to create HQ user with HQ role (rather than TOO) from devlocal login page
- HQ User has it's own queues page containing all other office queues and searches
- HQ User has ability to make the necessary requests for all those queues
- HQ User can view all pages but not edit anything
- HQ User can leave customer service remarks
- Test for HQ User's queues page, mores tests to be added in later tickets

> [!NOTE]  
> In the Files Changed tab, the bulk of the additions will be collapsed due to being large diffs.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Ensure the HQ role FF is enabled
2. Access the office application
3. Create a new HQ user in the KKFA GBLOC (Switching GBLOCs will come in a later ticket)
4. Ensure the following functionality is available:
- [x] You are able to view, filter, sort, and open moves from the TOO's Queue
- [x] You are able to view, filter, sort, and open moves from the TIO's Queue
- [x] You are able to view, filter, sort, and open moves from the SC's Counseling Queue
- [x] You are able to view, filter, sort, and open moves from the SC's PPM Closeout Queue
- [x] You are able to search for moves
- [x] You are able to search for customers
5. Ensure when you open a move from any of the queues or the move search you are able to see everything but edit nothing except customer support remarks.
6. From the customer search you should not be able to open any of the users because those opened to edit forms that didn't have an existing way to be made read only.
7. Stop the server, disable the HQ Role feature flag in `.envrc` and `development.features.yaml`, run `direnv allow`, and restart your server.
8. You should now be met with the Invalid Permissions page when you log in as your previously created HQ user.
9. Try to navigate to various pages by manually updating the the path (`/moves-queue`, `/move/SITFUT/details`, etc.) and ensure you are shown the same invalid permissions page.

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
